src/nvim/eval.c

```diff
@@ -10787,47 +10787,74 @@ static void f_jobresize(typval_T *argvars, typval_T *rettv)
   rettv->vval.v_number = 1;
 }
 
-// "jobstart()" function
-static void f_jobstart(typval_T *argvars, typval_T *rettv)
+static char **tv_to_argv(typval_T *cmd)
 {
-  rettv->v_type = VAR_NUMBER;
-  rettv->vval.v_number = 0;
+  if (cmd->v_type == VAR_STRING) {
+    return shell_build_argv((char *)get_tv_string(cmd), NULL);
+  }
 
-  if (check_restricted() || check_secure()) {
-    return;
+  if (cmd->v_type != VAR_LIST) {
+    EMSG2(_(e_invarg2), "expected String or List");
+    return NULL;
   }
 
-  if (argvars[0].v_type != VAR_LIST
-      || (argvars[1].v_type != VAR_DICT && argvars[1].v_type != VAR_UNKNOWN)) {
-    // Wrong argument types
-    EMSG(_(e_invarg));
-    return;
+  list_T *args = cmd->vval.v_list;
+  int argc = args->lv_len;
+  if (!argc) {
+    EMSG(_("Argument vector must have at least one item"));
+    return NULL;
   }
 
-  list_T *args = argvars[0].vval.v_list;
-  // Assert that all list items are strings
+  assert(args->lv_first);
+
+  const char_u *exe = get_tv_string_chk(&args->lv_first->li_tv);
+  if (exe == NULL || !os_can_exe(exe, NULL)) {
+    // String is not executable
+    if (exe) {
+      EMSG2(e_jobexe, exe);
+    }
+    return NULL;
+  }
+  
+  // Build the argument vector
+  int i = 0;
+  char **argv = xcalloc(argc + 1, sizeof(char *));
   for (listitem_T *arg = args->lv_first; arg != NULL; arg = arg->li_next) {
-    if (arg->li_tv.v_type != VAR_STRING) {
-      EMSG(_(e_invarg));
-      return;
+    char *a = (char *)get_tv_string_chk(&arg->li_tv);
+    if (a == NULL) {
+      // Did emsg in get_tv_string; just deallocate argv.
+      while (i-- > 0) {
+        xfree(argv[i]);
+      }
+      return NULL;
     }
+    argv[i++] = xstrdup(a);
   }
 
-  int argc = args->lv_len;
-  if (!argc) {
-    EMSG(_("Argument vector must have at least one item"));
+  return argv;
+}
+
+// "jobstart()" function
+static void f_jobstart(typval_T *argvars, typval_T *rettv)
+{
+  rettv->v_type = VAR_NUMBER;
+  rettv->vval.v_number = 0;
+
+  if (check_restricted() || check_secure()) {
     return;
   }
 
-  assert(args->lv_first);
- 
-  const char_u *exe = get_tv_string(&args->lv_first->li_tv);
-  if (!os_can_exe(exe, NULL)) {
-    // String is not executable
-    EMSG2(e_jobexe, exe);
+  if (argvars[1].v_type != VAR_DICT && argvars[1].v_type != VAR_UNKNOWN) {
+    // Wrong argument types
+    EMSG2(_(e_invarg2), "expected dictionary");
     return;
   }
 
+  char **argv = tv_to_argv(&argvars[0]);
```

Comment:
That simplified `termopen()`, but did you have something in mind for `jobstart()`?